src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -213,6 +213,11 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		}
 	}
 
+	// Gets language info depending on connection provider.
+	private getLanguageFlavor(providerName?: string): string {
+		return (providerName && providerName === 'KUSTO') ? 'kusto' : 'sql';
```

Comment:
Do you have any short-term plans to go in and try and fix all this so we don't have to have hardcoded checks like this scattered about? It makes maintenance and reliability a lot harder when we have to check all sorts of special cases...

Reply:
I discussed with Karl and implemented a solution to get this information from extension's package.json. WIll update the PR.